google/iam_kms_crypto_key.go

```diff
@@ -0,0 +1,84 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudkms/v1"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+var IamKmsCryptoKeySchema = map[string]*schema.Schema{
+	"crypto_key_id": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+}
+
+type KmsCryptoKeyIamUpdater struct {
+	resourceId string
+	Config     *Config
+}
+
+func NewKmsCryptoKeyIamUpdater(d *schema.ResourceData, config *Config) (ResourceIamUpdater, error) {
+	cryptoKey := d.Get("crypto_key_id").(string)
+	cryptoKeyId, err := parseKmsCryptoKeyId(cryptoKey, config)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing resource ID for for %s: %s", cryptoKey, err)
+	}
+
+	return &KmsCryptoKeyIamUpdater{
+		resourceId: cryptoKeyId.cryptoKeyId(),
+		Config:     config,
+	}, nil
+}
+
+func (u *KmsCryptoKeyIamUpdater) GetResourceIamPolicy() (*cloudresourcemanager.Policy, error) {
+	p, err := u.Config.clientKms.Projects.Locations.KeyRings.CryptoKeys.GetIamPolicy(u.resourceId).Do()
+
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving IAM policy for %s: %s", u.DescribeResource(), err)
+	}
+
+	cloudResourcePolicy := &cloudresourcemanager.Policy{}
+
+	err = Convert(p, cloudResourcePolicy)
```

Comment:
Consider extracting the conversion logic in a function and reuse it for the keys and the rings. See:
https://github.com/terraform-providers/terraform-provider-google/blob/master/google/iam_folder.go#L85